### PR TITLE
removes async and batchEnabled references

### PR DESCRIPTION
**Pull request by dmarvp**
This is for: https://app.clickup.com/t/3bp9dhh
and after completing: https://app.clickup.com/t/3bp9dh6

Changes:
- Removed any reference to `asyncMode`
- Removed any reference to `batchEnabled`
  - This included the "Goto and exit" section. 
- Updated the node minimal requirements
- Reordered the configuration section to move the Jest options **after** our regular `testing.json` options
- Added some missing options
- Removed the "bst test options" section to avoid confusion.
- Removed everything related to SMAPI
- Removed cardContent / cardTitle examples and mentions 
- Moved the available voices to their own section
- Some configuration options were removed only for simplification (ie: asyncE2EWaitInterval)

Things we might need to revisit:
- Goto and exit for the current implementation (is there a demand for it?)
- SMAPI
- Removed configuration parameters: should they have their own/internal place? should we clean the code associated with the ones we'll no longer use?
  - html
  - skillId
  - stage
  - type = simulation
  - batchEnabled
  - asyncMode
  - asyncE2EWaitInterval
  - virtualDeviceBaseURL

### CODE REVIEW

File: end-to-end/guide.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ ADDITIONALLY - we now support experimentally the SMAPI Simulation API. This can
 
 ## Installation
 ### Prerequisites
+* [Node.js v12 or superior](https://nodejs.org/en/download/)
```
Comment (jkelvie) [inline end-to-end/guide.md]: "or above"

File: end-to-end/guide.md [hunk 2]
```diff
@@ -94,34 +68,54 @@ An example `testing.json` file for end-to-end tests:
     "locales": "de-DE, en-US",
     "platform": "alexa",
     "type": "e2e",
-    "virtualDeviceToken": "<TOKEN>"
+    "virtualDeviceToken": "<VIRTUAL DEVICE TOKEN>"
 }
 ```
 
-Below the end-to-end testing configuration options and what they do are listed:
+The following are the different configuration options available ordered alphabetically:
```
Comment (jkelvie) [inline end-to-end/guide.md]: The following are the different configuration options available:

File: end-to-end/guide.md [hunk 3]
```diff
@@ -94,34 +68,54 @@ An example `testing.json` file for end-to-end tests:
...
-| type | The type of test being run - can be `unit`, `simulation`, or `e2e` - defaults to `unit` |
+| stt | Speech to text service to use on your test's utterances. Supported services are google and witai - defaults to google |
+| [trace](#viewing-response-payloads) | Causes request and response JSON payloads to be printed to the console |
+| traceOutput | Saves a copy of each request and response in the `test_output` folder for debugging purposes - defaults to false|
+| type | The type of test being run - can be `unit` or `e2e` - defaults to `unit` |
 | [virtualDeviceToken](/end-to-end/setup/) | For end-to-end tests that use virtual devices, this must be specified. [Get one here](../setup/) |
+| voiceId| For end-to-end tests that use spoken utterances, this must be specified. Sets the voice id for text to speech conversion of your utterances.|
 
-To override [Jest options](https://facebook.github.io/jest/docs/en/configuration.html), just set them under the "jest" key.
+Additionally, you can override [Jest options](https://facebook.github.io/jest/docs/en/configuration.html) under the "jest" key. Here are our default ones:
```
Comment (jkelvie) [inline end-to-end/guide.md]: "Here are the default settings:"

File: end-to-end/guide.md [hunk 4]
```diff
@@ -183,17 +158,17 @@ bst test --help
 
 ### Using environment variables as settings
 
-You can use environment variables inside the testing.json file by using the `${variable}` format.
+Speaking of environment variables, you can use them inside the `testing.json` file by using the `${variable}` format.
```
Comment (jkelvie) [inline end-to-end/guide.md]: "Regarding environment variables, you can reference them inside the `testing.json` file with the `${ENVIRONMENT_VARIABLE}` format"

File: end-to-end/guide.md [hunk 5]
```diff
@@ -926,17 +846,13 @@ The following are settings than can help you overcome specific testing issues. H
...
-| ignoreExternalErrors | When a not controlled error happens while executing your tests, marks them as skipped instead of failures - defaults to `false` |
-| [includeRaw](#batch-or-sequential-tests) | If true, the whole raw response from Alexa or Google is included as part of the response. This provides much more response details, but makes the response a lot bigger -  defaults to true |
-| [maxAsyncE2EResponseWaitTime](#batch-or-sequential-tests) | Set an interval in milliseconds to wait before stop looking for new results, when batchEnabled is set to false - defaults to 15000 |
+| ignoreExternalErrors | When a not controlled error happens while executing your tests, mark them as skipped instead of failures - defaults to `false` |
+| includeRaw | If true, the whole raw payload from the tested platform is included as part of the response. This provides much more insight, but makes the response a lot bigger -  defaults to true |
+| maxAsyncE2EResponseWaitTime | Set an interval in milliseconds to wait before stop looking for new results and throwing an error - defaults to 15000 |
```
Comment (jkelvie) [inline end-to-end/guide.md]: Can we alias this as "maxWaitTimeForResponse". I think this name is confusing and overly complicated (as well as reference async again)

**Review by jkelvie — CHANGES_REQUESTED** [review]

File: end-to-end/guide.md (hunk 5 above)
Comment (dmarvp) [reply end-to-end/guide.md]: I made the change in advance but 'll create a ticket for this.

**Review by jkelvie — APPROVED** [review]

**Pull request merged**